service/create/service.go

```diff
@@ -369,14 +394,37 @@ func (s *Service) runMachine(input runMachineInput) error {
 		return microerror.MaskAny(err)
 	}
 
+	clusterID := input.spec.Cluster.Cluster.ID
+	cloudconfigConfig := SmallCloudconfigConfig{
+		MachineType: input.prefix,
+		Region:      input.spec.AWS.Region,
+		S3Bucket:    s.s3Bucket,
+		ClusterID:   clusterID,
+	}
+
+	cloudconfigS3Path := fmt.Sprintf("%s/cloudconfig/%s",
+		cloudconfigConfig.ClusterID,
+		cloudconfigConfig.MachineType)
+	if err := s.uploadCloudconfigToS3(input.awsSession,
+		s.s3Bucket,
+		cloudconfigS3Path,
+		cloudConfig); err != nil {
+		return microerror.MaskAny(err)
+	}
+
+	smallCloudconfig, err := s.SmallCloudconfig(cloudconfigConfig)
+	if err != nil {
+		return microerror.MaskAny(err)
+	}
+
 	var reservation *ec2.Reservation
 	for i := 0; i < 5; i++ {
```

Comment:
It's just an arbitrary limit to stop trying to create the instance if we keep getting `Invalid IAM Instance Profile`. I can move it to a threshold constant/var.